nats.go

```diff
@@ -3289,14 +3288,17 @@ func (nc *Conn) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSyn
 			}
 		}
 
-		// for subscribe sync case we make it so users don't need to supply a inbox that they never use
-		if aopts.consumer.DeliverSubject == "" && subj == "" && (cb == nil || ch == nil) {
+		// when no DeliverySubject is given, while we don't have any support for pull mode
```

Comment:
I wonder if we remove deliver subject all together for the consumer config for the Go client?

Reply:
I have to send it to the server, and not all consumers are going to be unique inboxes based, I might want to do a queue subscribe on a js consumer between 2 clients - but want the first client that start to create the consumer rather than do so from cli/etc. then I have to do so on a well known delivery subject.

Reply:
Feels like that would fit under QueueSubscribe no? Trying to keep what folks already know top of mind.

Reply:
Regardless we can’t remove DeliverySubject even if we only do it in one place, I was responding specifically to you asking why not remove it.

Reply:
We need it sent to the server, but the public cfg struct that a user would utilize for the consumer config option for Subsc ribe() does not have to have it.

Reply:
So you suggest I make ConsumerConfig without that key and yet another type that, I guess, embeds it and has the subject that I then set, this 2nd shadow type is set to the server?

Tbh if I were to try learn how JS works this would just be all too magical if we have public structures that doesn’t match the documented APIs or schemas and where people need to read the nats.go code to figure out how the JS API works and what magical under the cover things happen- since it would be completely non obvious how you go from this API behaviour to how JetStream works - it would just be frustrating and confusing. 

As it is we are doing the absolute minimum here and by making it magic and non standard - the server API being standard - we are jus adding  a bunch of wtf behaviour and strange magic if we keep going this way. 

But that’s the last comment I will make so will just go ahead and do whatever you want so will do this next week.

Reply:
I think what we are trying to do is make a straightforward way for Go programmers to consume messages from a JetStream stream.

We both are pushing for the same thing I believe just approaching it differently. I would like JetStream users to see consumers as just another NATS subscription vs something totally different like we did with STAN. Under the covers they are different, but I believe at the base client lib these should just look like other subscribers.